arbnode/dataposter/slice/slicestorage.go

```diff
@@ -68,8 +68,8 @@ func (s *Storage) Put(_ context.Context, index uint64, prev, new *storage.Queued
 		s.queue = append(s.queue, new)
 	} else if index >= s.firstNonce {
 		queueIdx := int(index - s.firstNonce)
-		if queueIdx > len(s.queue) {
-			return fmt.Errorf("attempted to set out-of-bounds index %v in queue starting at %v of length %v", index, s.firstNonce, len(s.queue))
```

Comment:
any particular reason for suppressing this error? earlier out-of-bound issues will be masked as notEqual error now.

Also this might hide one off errors in s.queue. Please confirm with Lee once.

Reply:
Reverted this and addressed the part that was requiring out-of-bound indexes.